python/cudf/cudf/core/series.py

```diff
@@ -1505,9 +1505,14 @@ def _make_operands_and_index_for_binop(
                     "Can only compare identically-labeled Series objects"
                 )
             lhs, other = _align_indices([self, other], allow_non_unique=True)
-            can_use_self_column_name = self.name == other.name
         else:
             lhs = self
+
+        try:
+            can_use_self_column_name = cudf.utils.utils._is_same_name(
+                self.name, other.name
+            )
+        except Exception:
```

Comment:
This can only raise `AttributeError` from what I can tell.
```suggestion
        except AttributeError:
```